### PR TITLE
Add contributing section to README and contributor's guide

### PR DESCRIPTION
**Pull request by ali-ramadhan**
For JOSS we need to have a helpful `CONTRIBUTING.md` file, and I thought it would be good to put one up for Hacktoberfest.

I'm a big fan of MetPy's contributor's guide so I heavily based it on theirs, I don't think I could have done a better job.

MetPy uses a [BSD-3-Clause](https://tldrlegal.com/license/bsd-3-clause-license-(revised)) license so I think this means we'd have to license our `CONTRIBUTING.md` file under BSD-3-Clause? @christophernhill is this an issue?

cc MetPy devs @dopplershift @jrleeman: Your MetPy contributor's guide is really excellent! I hope you don't mind if I borrowed heavily from it.

### CODE REVIEW

**Comment by dopplershift** [comment]
We've borrowed heavily from other projects in putting that together, so by all means feel free to borrow! I have no idea how/if the license applies to a Contributor's guide.

File: README.md [hunk 1]
```diff
@@ -92,6 +92,13 @@ You can see some movies from GPU simulations below along with CPU and GPU [perfo
 ## Getting help
 If you are interested in using Oceananigans.jl or are trying to figure out how to use it, please feel free to ask us questions and get in touch! Check out the [examples](https://github.com/climate-machine/Oceananigans.jl/tree/master/examples) and [open an issue](https://github.com/climate-machine/Oceananigans.jl/issues/new) if you have any questions, comments, suggestions, etc.
 
+## Contributing
+If you're interested in contributing to the development of Oceananigans we want your help no matter how big or small a contribution you make! It's always great to have new people look at the code with fresh eyes: you will see errors that other developes have missed.
```
Comment (suyashbire1) [inline README.md]: typo: developes

**Comment by ali-ramadhan** [comment]
> We've borrowed heavily from other projects in putting that together, so by all means feel free to borrow! I have no idea how/if the license applies to a Contributor's guide.

Thank you!

Also thanks @suyashbire1 for catching the typos!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
+* Each pull request should consist of a logical collection of changes. You can
+  include multiple bug fixes in a single pull request, but they should be related.
+  For unrelated changes, please submit multiple pull requests.
+* Do not commit changes to files that are irrelevant to your feature or bugfix
+  (eg: .gitignore).
+* Be willing to accept criticism and work on improving your code; we don't want
+  to break other users' code, so care must be taken not to introduce bugs. We
+  discuss pull requests and keep working on them until we believe we've done a
+  good job.
+* Be aware that the pull request review process is not immediate, and is
```
Comment (glwagner) [inline CONTRIBUTING.md]: Should we add a statement that we encourage either commenting on an existing issue or opening a new one to discuss the problem being tackled prior to starting work on the PR?

**Pull request merged**